### PR TITLE
Changed the time-dependent behavior of sigma and learning rate

### PR DESCRIPTION
**Pull request by vkatariya8**
Changed the time constant for sigma. Changed the behavior of sigma and learning rate to an exponential decay which makes my current implementation of the SOM behave better.

### CODE REVIEW

**Comment by JustGlowing** [comment]
Hi vkatariya8 and thanks for your PR. I've checked your code and it seems okay. However, changing the update rule for `eta` and `sigma` you don't improve the behaviour of the algorithm in general. It may be better in some cases, but in other cases the rule currently improved may give better results.

**Comment by vkatariya8** [comment]
Hey

I understand. Maybe we can include an option to choose between the two behaviors?

**Comment by JustGlowing** [comment]
yes, that would be a possibility.

**Comment by vkatariya8** [comment]
Would you like me to implement that?

**Comment by JustGlowing** [comment]
Have fun if you want. Otherwise, I can implement this feature next weekend.

**Comment by vkatariya8** [comment]
I'll do it tomorrow. First pull request, I'm excited about it!



Sent using CloudMagic [https://cloudmagic.com/k/d/mailapp?ct=pa&cv=6.3.45&pv=5.1.1]

On Mon, Jul 13, 2015 at 9:36 pm, JustGlowing < notifications@github.com [notifications@github.com] > wrote:
Have fun if you want. Otherwise, I can implement this feature next weekend.

—
Reply to this email directly or view it on GitHub
[https://github.com/JustGlowing/minisom/pull/5#issuecomment-120981890] .[https://github.com/notifications/beacon/AHNFY-6rxdbxJWemLpg_oU1aM5lPqvGGks5oc9magaJpZM4FWA67.gif]

**Comment by vkatariya8** [comment]
Have added this option.

**Comment by JustGlowing** [comment]
Hi again vkatariya8, your solution was a bit outside of the style of the repository.

I added the functionality that you need in the master branch directly.

Now you can pass your custom decay function through the constructor. This example uses the function that you implemented:

```
som = MiniSom(7, 7, 4, sigma=1.0, learning_rate=0.5, decay_function=lambda x,t,T: x*exp(-t/T))
```

**Comment by vkatariya8** [comment]
Hello,

Cool. That's interesting, I'd never thought of passing the decay function as a parameter directly. Much more elegant.
